From b0324e84451e86e3b7a80346e132d8bbf6c3bdd5 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Wed, 7 Mar 2018 18:23:23 +0800
Subject: [PATCH 3441/5242] MLK-17569-1: hdp: fix channel swapping issue for
 hdmi audio

commit  9f9fcf0f7a6f226d8649367a09041d1090a292b4 from
https://source.codeaurora.org/external/imx/linux-imx.git

There is channel swapping issue for 4 channel and 8 channel audio.
After dump the register, found that SMPL2PKT_CNFG is not set
correctly, the reason is that F_NUM_OF_I2S_PORTS should be
F_NUM_OF_I2S_PORTS_S.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Reviewed-by: Sandor Yu <sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/hdp/API_Audio.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mxc/hdp/API_Audio.c b/drivers/mxc/hdp/API_Audio.c
index a09facb..2bfa0fc 100644
--- a/drivers/mxc/hdp/API_Audio.c
+++ b/drivers/mxc/hdp/API_Audio.c
@@ -108,7 +108,7 @@ CDN_API_STATUS CDN_API_AudioConfigCore(state_struct *state,
 		cdn_apb_write(state,
 			      ADDR_SOURCE_AIF_SMPL2PCKT + (SMPL2PKT_CNFG << 2),
 			      F_MAX_NUM_CH(numOfChannels - 1) |
-			      F_NUM_OF_I2S_PORTS((numOfChannels / 2) - 1) |
+			      F_NUM_OF_I2S_PORTS_S((numOfChannels / 2) - 1) |
 				  (1 << 8) | (lanesParam << 11));
 
 		if (numOfChannels == 2)
-- 
1.7.9.5

